### ISSUE TITLE
Can't set ApplicationStack.PythonVersion

### ISSUE THREAD

**Issue by FernandoMorais** [pulumi/pulumi-azure#1227]
### What happened?

I'm trying to set the PythonVersion on Azure.AppService.LinuxFunctionApp and getting conflicting configuration errors.

```
var exampleLinuxFunctionApp = new Azure.AppService.LinuxFunctionApp("exampleLinuxFunctionApp", new()
    {
        ResourceGroupName = exampleResourceGroup.Name,
        Location = exampleResourceGroup.Location,
        StorageAccountName = exampleAccount.Name,
        StorageAccountAccessKey = exampleAccount.PrimaryAccessKey,
        ServicePlanId = exampleServicePlan.Id,
        SiteConfig = new AppServiceInputs.LinuxFunctionAppSiteConfigArgs 
        {
                ApplicationStack = new AppServiceInputs.LinuxFunctionAppSiteConfigApplicationStackArgs
                {
                        PythonVersion = "3.9",
                }
        },
    });
```

### Expected Behavior

The function app is created as expected.

### Steps to reproduce

Follow the example provided by the documentation page using C# language.
https://www.pulumi.com/registry/packages/azure/api-docs/appservice/linuxfunctionapp/#example-usage

**Set the PythonVersion:**
```
[...]
SiteConfig = new AppServiceInputs.LinuxFunctionAppSiteConfigArgs 
{
        ApplicationStack = new AppServiceInputs.LinuxFunctionAppSiteConfigApplicationStackArgs
        {
                PythonVersion = "3.9",
        }
}
[...]
```

**You'll get the error:**

  azure:appservice:LinuxFunctionApp (exampleLinuxFunctionApp):
    error: azure:appservice/linuxFunctionApp:LinuxFunctionApp resource 'exampleLinuxFunctionApp' has a problem: Conflicting configuration arguments: "site_config.0.application_stack.0.use_dotnet_isolated_runtime": conflicts with site_config.0.application_stack.0.python_version. Examine values at 'LinuxFunctionApp.SiteConfig.ApplicationStack[0].UseDotnetIsolatedRuntime'.

### Output of `pulumi about`

CLI
Version 3.55.0
Go Version go1.20
Go Compiler gc

Plugins
NAME VERSION
dotnet unknown

Host
OS darwin
Version 13.1
Arch arm64

This project is written in dotnet: executable='/usr/local/share/dotnet/dotnet' version='7.0.102'

[...]

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by rquitales** [comment]
Thanks for reporting this. I am able to reproduce this issue with C# as well as other languages such as Typescript. This would appear to be a bug in our provider and we'll look into fixing this!

**Comment by rquitales** [comment]
Looks like an issue related to https://github.com/pulumi/pulumi-terraform-bridge/issues/577. In the meantime, I can patch our fork to fix this issue until this upstream bug is resolved.

**Comment by rquitales** [comment]
@FernandoMorais I have patched this in #1226 and this should be rolling out in our next release (happening as we speak). Please update your SDK and provider to use v5.36.0 when it is official in the next few hours. Thanks!
